src/init/common.cpp

```diff
@@ -94,6 +96,26 @@ void SetLoggingOptions(const ArgsManager& args)
     fLogIPs = args.GetBoolArg("-logips", DEFAULT_LOGIPS);
 }
 
+void SetLoggingLevel(const ArgsManager& args)
+{
+    if (args.IsArgSet("-loglevel")) {
+        const auto loglevels = args.GetArgs("-loglevel");
+        for (const auto& loglevel : loglevels) {
+            const auto is_global = loglevel.find(':') == std::string::npos;
+            if (is_global) {
+                if (!LogInstance().SetLogLevel(loglevel)) {
+                    InitWarning(strprintf(_("Unsupported global threshold logging level provided -loglevel=%s. Valid values: debug, none, info, warning, error."), loglevel));
+                };
+            } else {
+                const auto toks = SplitString(loglevel, ':');
+                if (!(toks.size() == 2 && LogInstance().SetCategoryLogLevel(toks[0], toks[1]))) {
+                    InitWarning(strprintf(_("Unsupported category specific threshold logging level provided -loglevel=%s. Expected -loglevel=<category>:<loglevel>."), loglevel));
```

Comment:
here and elsewhere, missing hyphen `s/category specific/category-specific/` and "threshold" seems unneeded, could be omitted
```suggestion
                    InitWarning(strprintf(_("Unsupported category-specific logging level provided: -loglevel=%s. Expected -loglevel=<category>:<loglevel>."), loglevel));
```

Reply:
Done 👍🏻